lib/orocos/process.rb

```diff
@@ -1142,92 +1165,91 @@ def spawn(log_level: nil, working_directory: Orocos.default_working_directory,
                           elsif wait
                               Float::INFINITY
                           end
-                wait_running(timeout, name_service)
+                wait_running(timeout)
             end
         end
 
-        def self.resolve_all_tasks(process, cache = Hash.new)
```

Comment:
Keep the `cache` argument, just renaming it into `resolved_tasks` and not defining `resolved_tasks` as a local variable. Just to keep the current behavior.